### DISCUSSION TITLE
User default values

### DISCUSSION THREAD

**Discussion by automatorz** [tteck/Proxmox#3796] [Ideas]
It would be cool if the scripts would allow the user to specify his own defaults, which would override the script defaults.
Imagine the following scenario(s):
- You have to deploy 10 debian LXC containers, using the default values, except you want 2GB of RAM instead of the default 512MB
- You have to deploy 10 debian machines, using the default values, except you want to set a custom MAC address and root password

Currently the only way to do that, is to choose an advanced installation, and then go through all of the settings choosing the default value, except for the 1-2-3 values that you actually want to change.

It would be nice if one could use env variables to specify their own defaults:
```
export HS_RAM_SIZE=2 # HS prefix for Helper Scripts
export HS_CORE_COUNT=2 # HS prefix for Helper Scripts
export HS_DISK_SIZE=20 # HS prefix for Helper Scripts
...
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/debian.sh)"
```

and then the script could do something like:
```
VAR_RAM="${HS_RAM_SIZE:-512}"
```

This way, the script sets a default value, but allows it to be overridden by the user, thus easing installs of multiple machines.

It would also be nice if setting one of the variables, e.g.:
```
export HS_ACKNOWLEDGE_SCRIPT_RISK=YES
export HS_NO_ADVANCED_QUESTIONS=YES
```

one could for example also skip the other question pop-ups (e.g. "This will create a New Debian LXC. Proceed?" or "Use Default Settings?"), thus further easing automations, or even adding the ability to create wrapper scripts (e.g. setting up a 10-node kubernetes cluster, using LXC containers provisioned by helper-scripts).

**Comment by tteck** [comment]
Not interested, sorry

**Comment by automatorz** [comment]
Is there a specific reason why you're not interested? If I were to create the pull requests for the needed code changes, would you be willing to look into them?
I mean functionality wise the current way to run the scripts would not be affected, and also from the maintenance perspective, I don't think there would be much of an impact either...
